### PR TITLE
feat: Add improvements to populate_product_catalog_command

### PR DESCRIPTION
**Pull request by AfaqShuaib09**
[PROD-4167](https://2u-internal.atlassian.net/browse/PROD-4167)
------

This PR adds following improvements to management command
- Skip record if required detail is empty.
- Remove duplicates from language_code method.
- Add flag to include inactive runs' language codes in the method.
- Fix N+1 query issue.
- Remove dashes from uuid.

### CODE REVIEW

File: course_discovery/apps/course_metadata/tests/factories.py [hunk 1]
```diff
@@ -669,6 +669,7 @@ class Meta:
     type = factory.SubFactory(ProgramTypeFactory)
     marketing_slug = factory.Sequence(lambda n: f'test-slug-{n}')
     card_image_url = FuzzyText(prefix='https://example.com/program/card')
+    card_image = factory.django.ImageField()
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/tests/factories.py]: why is this needed here?

File: course_discovery/apps/course_metadata/management/commands/populate_product_catalog.py [hunk 2]
```diff
@@ -88,6 +88,10 @@ def get_products(self, product_type, product_source):
             if product_source:
                 queryset = queryset.filter(product_source__slug=product_source)
 
+            queryset = queryset.annotate(
+                num_orgs=Count('authoring_organizations')
+            ).filter(Q(num_orgs__gt=0) & Q(image__isnull=False) & ~Q(image=''))
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/management/commands/populate_product_catalog.py:93]: is this image check against organization or against course? There are duplicate fields, so you might want to confirm which one is not deprecated.

File: course_discovery/apps/course_metadata/tests/factories.py (hunk 1 above)
Comment (AfaqShuaib09) [reply course_discovery/apps/course_metadata/tests/factories.py]: Previously, it was not added here. I added it now so I don't have to explicitly define the image value when creating a program or degree object through the factory

File: course_discovery/apps/course_metadata/management/commands/populate_product_catalog.py (hunk 2 above)
Comment (AfaqShuaib09) [reply course_discovery/apps/course_metadata/management/commands/populate_product_catalog.py:93]: against course, I've verified it from admin. It's not deprecated

File: course_discovery/apps/course_metadata/tests/factories.py (hunk 1 above)
Comment (AfaqShuaib09) [reply course_discovery/apps/course_metadata/tests/factories.py]: However, it's failing other test cases, so I think I need to initialize it directly in the test case where it's required.

File: course_discovery/apps/course_metadata/tests/test_models.py [hunk 3]
```diff
@@ -100,6 +100,30 @@ def test_image_url(self):
...
+            language=LanguageTag.objects.get(code='en')
+        )
+        SeatFactory.create(course_run=course_run)
+        CourseRunFactory(
+            course=Course.objects.all()[0],
+            status=CourseRunStatus.Unpublished,
+            language=LanguageTag.objects.get(code='es'),
+            enrollment_end=datetime.datetime.now() - datetime.timedelta(days=5)
+        )
+        assert course.languages_codes() in ['en, es', 'es, en']
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/tests/test_models.py]: isn't es, en and en, es the same thing?

Comment (DawoudSheraz) [reply course_discovery/apps/course_metadata/tests/test_models.py]: What you can do is convert the set back to list and sort in a particular order. The changing order of the data with every hit is not a consistent behavior

Comment (AfaqShuaib09) [reply course_discovery/apps/course_metadata/tests/test_models.py]: sure, lemme update this

**Review by DawoudSheraz — APPROVED** [review]

File: course_discovery/apps/course_metadata/management/commands/populate_product_catalog.py [hunk 4]
```diff
@@ -106,6 +110,10 @@ def get_products(self, product_type, product_source):
             if product_source:
                 queryset = queryset.filter(product_source__slug=product_source)
 
+            queryset = queryset.annotate(
+                num_orgs=Count('authoring_organizations')
+            ).filter(Q(num_orgs__gt=0) & Q(card_image__isnull=False) & ~Q(card_image=''))
```
Comment (Ali-D-Akbar) [inline course_discovery/apps/course_metadata/management/commands/populate_product_catalog.py:115]: why are we filtering out based on `card_image` here while we used `image` before?

**Review by Ali-D-Akbar — APPROVED** [review]
one comment from me

File: course_discovery/apps/course_metadata/management/commands/populate_product_catalog.py (hunk 4 above)
Comment (AfaqShuaib09) [reply course_discovery/apps/course_metadata/management/commands/populate_product_catalog.py:115]: It's for degrees while the above one is for courses
